src/middlewared/middlewared/service.py

```diff
@@ -624,6 +624,38 @@ async def generate_locked_alert(self, share_task_id):
             {**share_task, 'identifier': await self.human_identifier(share_task), 'type': self.share_task_type}
         )
 
+    @private
+    def datasets_in_use(self, share_task_id, datasets=None):
```

Comment:
It seems to be we are starting to overcomplicate this service base, just for some obscure use case?

Reply:
How about libzfs.get_dataset_by_path ?